### PR TITLE
feat: add sign up form from soul

### PR DESCRIPTION
**Pull request by apledger**
## What/Why?
Add a `DynamicForm` component, all related form fields, and the dynamic `SignUpSection`.

### Success
https://github.com/user-attachments/assets/8ddf3aec-b478-4379-94c7-536f9ec028fe

### Client side validation
https://github.com/user-attachments/assets/b008b1c8-7496-4776-871d-72767bdc940e

### Custom required field
https://github.com/user-attachments/assets/3fdb6fd8-6ee9-42f7-9e69-8ade9cb769f1

### CODE REVIEW

File: core/app/[locale]/(default)/(auth)/register/page.tsx [hunk 1]
```diff
@@ -26,18 +32,25 @@ export default async function Register() {
...
-      <RegisterCustomerForm
-        addressFields={addressFields}
-        customerFields={customerFields}
-        reCaptchaSettings={reCaptcha}
-      />
-    </div>
+    <SignUpSection
+      action={registerCustomer}
+      fields={[
+        addressFields
```
[the remark this answers is not in the document]
Comment (apledger) [reply core/app/[locale]/(default)/(auth)/register/page.tsx:42]: Nope because we want them grouped on the same line.

File: core/app/[locale]/(default)/(auth)/register/page.tsx [hunk 2]
```diff
@@ -26,18 +32,25 @@ export default async function Register() {
...
-      <h1 className="my-6 text-4xl font-black lg:my-8 lg:text-5xl">{t('heading')}</h1>
-      <RegisterCustomerForm
-        addressFields={addressFields}
-        customerFields={customerFields}
-        reCaptchaSettings={reCaptcha}
-      />
-    </div>
+    <SignUpSection
+      action={registerCustomer}
+      fields={[
```
[the remark this answers is not in the document]
Comment (apledger) [reply core/app/[locale]/(default)/(auth)/register/page.tsx:41]: Sure, I just need a form skeleton then. Do we have any?

[the remark this answers is not in the document]
Comment (apledger) [reply core/app/[locale]/(default)/(auth)/register/page.tsx:41]: And could we maybe save that for a future PR? That seems like scope we could potentially cut seeing as how main currently doesn't support this.

[the remark this answers is not in the document]
Comment (jorgemoya) [reply core/app/[locale]/(default)/(auth)/register/page.tsx:41]: We've been building this functionality as part of the release by default, having said that we can track this in a different issue, unless @migueloller feel otherwise.

**Review by jorgemoya — APPROVED** [review]
Seems good to me but I would like for @migueloller to also take a look at this before we merge.

File: core/app/[locale]/(default)/(auth)/register/page.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (migueloller) [reply core/app/[locale]/(default)/(auth)/register/page.tsx:41]: I think that if we make it streamable we should make a skeleton component to use as fallback. We could start without streamable support, which would make this page have to `await` the fields in the server, increasing TTFB, which isn't a big deal for a sign up page. That being said, I still think we should have a skeleton for the dynamic form so we _can_ make fields streamable and be consistent

File: core/app/[locale]/(default)/(auth)/register/_actions/register-customer.ts [hunk 3]
```diff
@@ -34,72 +39,156 @@ const RegisterCustomerMutation = graphql(`
...
+        ].includes(field.name),
+    );
+
+  return {
+    email: String(value[FieldNameToFieldId.email] ?? ''),
+    password: String(value[FieldNameToFieldId.password] ?? ''),
+    firstName: String(value[FieldNameToFieldId.firstName] ?? ''),
+    lastName: String(value[FieldNameToFieldId.lastName] ?? ''),
+    phone: String(value[FieldNameToFieldId.phone] ?? ''),
+    company: String(value[FieldNameToFieldId.company] ?? ''),
```
Comment (migueloller) [inline core/app/[locale]/(default)/(auth)/register/_actions/register-customer.ts:67]: Is it appropriate for these to be empty strings instead of `null` or something else? I'm not familiar enough with the BigCommerce GraphQL API to know (same applies to other instances of this pattern below)

**Review by migueloller — CHANGES_REQUESTED** [review]
Looking good! Requesting changes since there's still some downstream dependencies from https://github.com/makeswift/vibes/pull/370

This is the sort of component that has so many edge cases and is so complex that I think it deserves a robust E2E test suite. Something for us to invest in soon after NRF @chanceaclark, @jorgemoya, and @deini

File: core/app/[locale]/(default)/(auth)/register/page.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (apledger) [reply core/app/[locale]/(default)/(auth)/register/page.tsx:41]: Agreed that using a skeleton and streaming is a better UX, I just think that since the baseline main version doesn't have this behavior we should prioritize it last given all of the other work we need to get through.

[the remark this answers is not in the document]
Comment (migueloller) [reply core/app/[locale]/(default)/(auth)/register/page.tsx:41]: Yeah, I agree. We can start without streamable props support and without a skeleton for now and introduce that later.

[the remark this answers is not in the document]
Comment (jorgemoya) [reply core/app/[locale]/(default)/(auth)/register/page.tsx:41]: I will track this because I think we definitely want to include.

[the remark this answers is not in the document]
Comment (migueloller) [reply core/app/[locale]/(default)/(auth)/register/page.tsx:41]: Sounds good. I think at this point given how close we are to the deadline, we're focused on feature parity with `main` and not wanting to spend much time on new functionality or behaviors not in `main`. Since `main` doesn't stream or have a skeleton for the sign up form, it's ok for us to ship it without it. We all agree that we should track it to include eventually, though!

File: core/app/[locale]/(default)/(auth)/register/_actions/register-customer.ts (hunk 3 above)
Comment (apledger) [reply core/app/[locale]/(default)/(auth)/register/_actions/register-customer.ts]: Refactored to parsing with zod.

**Review by migueloller — CHANGES_REQUESTED** [review]
Looks like the branch is out of date 😞 

Needs a rebase before I can re-review

**Review by migueloller — APPROVED** [review]
